ODIN_II/SRC/include/BlockMemories.hpp

```diff
@@ -21,25 +21,26 @@
  * WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
  * FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
  * OTHER DEALINGS IN THE SOFTWARE.
+ *
+ * @file This file includes the defintion of basic structure used 
+ * in Odin-II Block Memory resolving process. Moreover, it provides
+ * the declaration of the related public routines
  */
 #ifndef _BLOCK_MEMORIES_H_
 #define _BLOCK_MEMORIES_H_
 
 #include <unordered_map>
 
-extern vtr::t_linked_vptr* block_memory_list;
-extern vtr::t_linked_vptr* read_only_memory_list;
-
-const int REG_INFERRENCE_THRESHOLD_WIDTH = 32;
-const int REG_INFERRENCE_THRESHOLD_DEPTH = 20;
-const int LUTRAM_MAX_THRESHOLD_AREA = 2560;
-const int LUTRAM_MIN_THRESHOLD_AREA = 640;
+const int REG_INFERRENCE_THRESHOLD_WIDTH = 32; // Width threshold for register array inference
+const int REG_INFERRENCE_THRESHOLD_DEPTH = 20; // Depth threshold for register array inference
+const int LUTRAM_MAX_THRESHOLD_AREA = 2560;    // Maximum area threshold for LUTRAM inference
+const int LUTRAM_MIN_THRESHOLD_AREA = 640;     // Minimum area threshold for LUTRAM inference
```

Comment:
Probably this number should be lower.  While a 6-LUT based 10 BLE/logic block architecture wastes some bits for a LUTRAM smaller than 640 bits, it is still a much smaller implementation than using registers.  An Intel architecture with such a logic block can only implement 40 registers for example, so the LUTRAM min inference should probably be more like 80.

This comment should also say the units of LUTRAM_MIN_THRESHOLD_AREA (bits in a memory array I assume?).

INFERRENCE is mispelt in the first two const int s: it should be INFERENCE
I think that's worth correcting as it means find operations won't work unless someone knows to misspell it again.

Reply:
You are absolutely right. So, considering each ALM with 6-LUT and 4 registers, one LAB(10 ALMs) can implement 40 bits registers or 640 bits LUTRAM.  That makes sense to consider the reg inference threshold as the reg size of two LABs with local interconnect [link](https://www.intel.com/content/www/us/en/programmable/documentation/wtw1441782332101.html#wtw1443169713037). Then, the maximum threshold for the REGISTER INFERENCE would be 80 bits which is the minimum threshold for LUTRAM INFERENCE as well. 
To wrap up :
1) Less than 80 bits => can be inferred as registers
2) 80 - 640 bits => better to be implemented using LUTRAM
3) more than 640 bits ( if depth > 64) => better to use M20k to avoid muxing

code:
```
const int REG_INFERENCE_THRESHOLD_MAX = 80;       // Max number of bits for register of array inference
const int LUTRAM_INFERENCE_THRESHOLD_MIN = 80;    // Max number of bits for LUTRAM inference
const int LUTRAM_INFERENCE_THRESHOLD_MAX = 640;   // Minimum area threshold for LUTRAM inference
```
Would you please correct me if I am wrong?